pkg/querier/astmapper/shard_summer_test.go

```diff
@@ -0,0 +1,79 @@
+package astmapper
+
+import (
+	"fmt"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestShardSummer(t *testing.T) {
+	var testExpr = []struct {
+		shards   int
+		input    string
+		expected string
+	}{
+		{
+			3,
+			`sum by(foo) (rate(bar1{baz="blip"}[1m]))`,
+			`sum by(foo) (
+			  sum by(foo) (rate(bar1{__cortex_shard__="0_of_3",baz="blip"}[1m])) or
+			  sum by(foo) (rate(bar1{__cortex_shard__="1_of_3",baz="blip"}[1m])) or
+			  sum by(foo) (rate(bar1{__cortex_shard__="2_of_3",baz="blip"}[1m]))
+			)`,
+		},
+		{
+			2,
+			`sum(
+				sum by (foo) (rate(bar1{baz="blip"}[1m]))
+				/
+				sum by (foo) (rate(foo{baz="blip"}[1m]))
+			)`,
+			`sum(
+			  sum by(foo) (
+				sum by(foo) (rate(bar1{__cortex_shard__="0_of_2",baz="blip"}[1m])) or
+				sum by(foo) (rate(bar1{__cortex_shard__="1_of_2",baz="blip"}[1m]))
+			  )
+			  /
+			  sum by(foo) (
+				sum by(foo) (rate(foo{__cortex_shard__="0_of_2",baz="blip"}[1m])) or
+				sum by(foo) (rate(foo{__cortex_shard__="1_of_2",baz="blip"}[1m]))
+			  )
+			)`,
+		},
+		// This is currently redundant: sums split into sharded versions, including summed sums.
+		{
+			2,
+			`sum(sum by(foo) (rate(bar1{baz="blip"}[1m])))`,
```

Comment:
can you add example with `sum without()` and `sum(rate(bar1{baz="blip"}[1m]))` ?

Reply:
added in b892aca